README.rst

```diff
@@ -582,6 +582,15 @@ in an integer). For example::
     nproc:
       type: int
 
+Interaction with containers: the MPIRequirement currently prepends its
+commands to the front of the command line that is constructed. If you
+wish to run a containerised application in parallel, for simple use
+cases this does work with Singularity, depending upon the platform
+setup. However this combination should be considered "alpha" -- please
+do report any issues you have! This does not work with Docker at the
+moment. (More precisely, you get `n` copies of the same single process
+image run at the same time that cannot communicate with each other.)
```

Comment:
Thanks for the information. If it doesn't work with Docker then we should not try to run if `DockerRequirement` is under `requirements` and we are not in `--singularity` mode. Likewise if `DockerRequirement` is under `hints` and we have neither `--no-containers` nor `--singularity`.

Did you test with udocker?

Reply:
Due to an ongoing issue with our cluster, I've not been able to test with udocker, however documentation suggests that it *should* work with MPI. Someone else testing this might be a useful exercise (of this and the wider MPIRequirement feature).

I have added the necessary code (and some tests) to stop if one has both Docker and MPI in `requirements` (same for `hints` as no obvious tie-breaker). If one is hinted and the other required, then the hint is dropped (and warned)